src/modules/vendor/components/courses/CourseContainer.vue

```diff
@@ -1,27 +1,59 @@
 <template>
   <div class="course-container">
     <div class="text-weight-bold q-mb-sm">{{ title }}</div>
-    <course-card class="q-mb-sm" v-for="(course, index) in courses" :key="index" :course="course"
+    <course-card class="q-mb-sm" v-for="(course, index) in courseSorted" :key="index" :course="course"
       @click="goToCourseProfile" />
   </div>
 </template>
 
 <script>
 import CourseCard from 'src/modules/vendor/components/courses/CourseCard';
+import { courseMixin } from 'src/modules/vendor/mixins/courseMixin';
 
 export default {
   name: 'CourseContainer',
+  mixins: [courseMixin],
   props: {
     title: { type: String, default: '' },
     courses: { type: Array, default: () => [] },
   },
   components: {
     'course-card': CourseCard,
   },
+  data () {
+    return {
+      courseSorted: [],
+    };
+  },
+  watch: {
+    courses () {
+      this.refreshCourseSorted();
+    },
+  },
   methods: {
     goToCourseProfile (course) {
       this.$router.push({ name: 'profile course info', params: { courseId: course._id } });
     },
+    getRangeToNearestDate (course) {
+      if (course.slots.length === 0) return Number.MAX_SAFE_INTEGER;
+
+      if (this.title === 'À venir') {
+        const firstSlot = course.slots[0];
+        return this.$moment(firstSlot[0].startDate).diff(this.$moment(), 'd', true);
+      } else if (this.title === 'Terminée(s)') {
+        const lastSlot = course.slots[course.slots.length - 1];
+        return this.$moment().diff(this.$moment(lastSlot[0].startDate), 'd', true);
+      } else if (this.title === 'En cours') {
+        const nextSlot = course.slots.filter((daySlots) => !this.happened(daySlots))[0];
+        return this.$moment(nextSlot[0].startDate).diff(this.$moment(), 'd', true);
+      }
+    },
+    refreshCourseSorted () {
+      this.courseSorted = this.courses.sort((a, b) => this.getRangeToNearestDate(a) - this.getRangeToNearestDate(b));
```

Comment:
tu pourrais faire une computed `courseSorted` et ainsi éviter un watcher. par contre, il faudra `cloneDeep` (lodash) this.courses pour éviter de le modifier.